### ISSUE TITLE
Schema mismatch may form during rolling upgrade from 1.7.x to 2.0 with more than 2 nodes

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2802]
*Installation details*
Scylla version: 2.0+
Cluster size: >= 3

When there are at least 2 nodes upgraded to 2.0, and the two exchanged schema for some reason, reads or writes which involve both 1.7 and 2.0 nodes may start to fail with the following error logged:
```
ERROR 2017-09-14 10:41:21,835 127.0.0.1   [shard 0] storage_proxy - Exception when communicating with 127.0.0.3: Failed to load schema version 58fc9b89-74ab-37ca-8640-8b38a1204f8d
```

The situation should heal after whole cluster is upgraded.

Table schema versions are calculated by 2.0 nodes differently than 1.7 nodes due to change in the schema tables format. Mismatch is meant to be avoided by having 2.0 nodes calculate the old digest on schema migration during upgrade, and use that version until next time the table is altered. It is thus not allowed to alter tables during the rolling upgrade.

Two 2.0 nodes may exchange schema, if they detect through gossip that their schema versions don't match. They may not match temporarily during boot, until the upgraded node completes booting and propagates its new schema through gossip. Such schema pull will result in a schema merge, which due to a bug will consider all affected tables as altered. This will cause recalculation of table schema versions, and cause mismatch with 1.7 nodes.

**Comment by glommer** [comment]
@tgrabiec let's just get our terms right: by bootstrap, what do you mean ? There is no streaming here.

**Comment by tgrabiec** [comment]
@glommer It should read "booting" (corrected)

**Comment by amoskong** [comment]
Verified that the issue was fixed in `scylla-2.0.rc5-0.20170917.0b49cfc.el7.centos.x86_64.rpm`.
